src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/LimitedPartnerController.java

```diff
@@ -46,46 +46,71 @@ public ResponseEntity<LimitedPartnerSubmissionCreatedResponseDto> createLimitedP
             @RequestAttribute(TRANSACTION_KEY) Transaction transaction,
             @Valid @RequestBody LimitedPartnerDto limitedPartnerDto,
             @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId,
-            @RequestHeader(value = ERIC_IDENTITY) String userId) {
+            @RequestHeader(value = ERIC_IDENTITY) String userId)
+    throws MethodArgumentNotValidException {
 
         var transactionId = transaction.getId();
         var logMap = new HashMap<String, Object>();
         logMap.put(URL_PARAM_TRANSACTION_ID, transactionId);
-
+        ApiLogger.infoContext(requestId, "Create a Limited Partner submission", logMap);
         try {
-            ApiLogger.infoContext(requestId, "Create a Limited Partner submission", logMap);
-
-            var submissionId = limitedPartnerService.createLimitedPartner(transaction, limitedPartnerDto, requestId,
-                    userId);
-
+            String submissionId = limitedPartnerService.createLimitedPartner(transaction, limitedPartnerDto, requestId, userId);
             var location = URI.create(String.format(URL_GET_LIMITED_PARTNER, transactionId, submissionId));
-
             var response = new LimitedPartnerSubmissionCreatedResponseDto(submissionId);
             return ResponseEntity.created(location).body(response);
-
         } catch (ServiceException e) {
+            ApiLogger.errorContext(requestId, "ServiceException occurred while creating Limited Partner", e, logMap);
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (NoSuchMethodException e) {
```

Comment:
for these 2 exceptions they seem to be handled the same so I would just have the one message and return code.
like this:
`} catch (ServiceException | NoSuchMethodException e) {
      ApiLogger.errorContext(requestId, "Error creating Limited Partner", e, logMap);
      return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
  }
`